### PR TITLE
Gh 1841 batch patches

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
When calling `updateBlocks` in the mutator.ts. The code was sending these to the server in separate `updateBlock` methods. `updateBlock` employed transactions and occasionally with SQLite, the second update would be called before completion of the first transaction. This was causing an error with SQLIte and a partial update. 

This PR moves `patchBlocks` to the server and places the entire thing into a transaction. In addition, it updates `insertBlocks` on the server to place the insert for all blocks in a single transaction.

#### Ticket Link
  Fixes https://github.com/mattermost/focalboard/issues/1841

### CODE REVIEW

**Review by ogi-m — APPROVED** [review]
Tested and passed
- working as expected now
- tested Personal Desktop and plugin

File: server/api/api.go [hunk 1]
```diff
@@ -676,6 +677,76 @@ func (a *API) handlePatchBlock(w http.ResponseWriter, r *http.Request) {
...
+	err = json.Unmarshal(requestBody, &patches)
+	if err != nil {
+		a.errorResponse(w, r.URL.Path, http.StatusInternalServerError, "", err)
+		return
+	}
+
+	auditRec := a.makeAuditRecord(r, "patchBlocks", audit.Fail)
+	defer a.audit.LogRecord(audit.LevelModify, auditRec)
+
+	// stampModificationMetadata(r, patches.BlockIDs, auditRec)
```
Comment (wiggin77) [inline server/api/api.go:736]: Should this just be removed?

File: server/model/block.go [hunk 2]
```diff
@@ -95,6 +95,17 @@ type BlockPatch struct {
 	DeletedFields []string `json:"deletedFields"`
 }
 
+// BlockPatchBatch is a batch of IDs and patches for modify blocks
+// swagger:model
+type BlockPatchBatch struct {
+	// The id for this block's parent block. Empty for root blocks
```
Comment (wiggin77) [inline server/model/block.go:101]: Are these really the parent ids?

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Just two possible comments to modify.

**Pull request merged**